### DISCUSSION TITLE
Setting sheldon config file is not enough

### DISCUSSION THREAD

**Discussion by seqizz** [rossmacarthur/sheldon#155] [Q&A] [answered]
Hi there :wave: 

I've upgraded to 0.7.0, and apart from a small change I needed to do in config.toml, sheldon failed to start because of non-existent config dir:

```
error: failed to acquire lock on config directory
  due to: failed to open `/home/gurkan/.config/sheldon`
  due to: No such file or directory (os error 2)
```

Just wanted to ask if this is intended, or it's because my setup is manual? Initially it didn't make sense to need a config dir while `SHELDON_CONFIG_FILE` already given, but for locking need I created the mentioned folder and it works ok now.

This looks like it could benefit a new parameter like "SHELDON_LOCK_FILE" which is mutually exclusive with config dir, for clarity.

**Comment by rossmacarthur** [comment]
So the one problem was in the 0.7.0 release we introduced the profiles feature which uses a separate lock file for each profile. How would you imagine this setting working in conjunction with that feature? 🤔

**Reply by seqizz** [reply]
Oh I wasn't aware of profiles, now after reading release notes, cool stuff :eyes: 

Assuming this locking is done per profile (if not, one parameter would be enough already), what about placing this temporary lock to `SHELDON_DATA_DIR` :thinking:

**Reply by rossmacarthur** [reply]
Actually rereading your initial question I think I might misunderstood your original issue. I think this is a bug.

**Comment by rossmacarthur** [comment, marked as answer]
This was a bug that fixed in a632fc59ead474c07d21dd8980e7c97364350ebb and released in 0.7.1 (a while ago now!)
